dimod/generators/bpsp.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import dimod
+import numpy.random as rnd
+
+__all__ = ["random_binary_paint_shop_problem"]
+
+
+def binary_paint_shop_problem(car_sequence):
```

Comment:
We aren't consistent everywhere in dimod (or Ocean) but for new stuff we try to use [typing](https://docs.python.org/3/library/typing.html). So
```suggestion
def binary_paint_shop_problem(car_sequence: typing.List[typing.Hashable]) -> dimod.BinaryQuadraticModel:
```
here and elsewhere